sdk/cosmos/src/requests/create_slug_attachment_builder.rs

```diff
@@ -94,7 +95,7 @@ where
 }
 
 impl<'a, 'b> CreateSlugAttachmentBuilder<'a, 'b, Yes, Yes> {
-    pub async fn execute(&self) -> Result<CreateSlugAttachmentResponse, CosmosError> {
+    pub async fn execute(self) -> Result<CreateSlugAttachmentResponse, CosmosError> {
```

Comment:
Same here, if it is used many times, this should be changes back to `&self`.

Reply:
Generally speaking, the idea of taking `&self` in execute is to make it easier to implement retry (Azure services can be offline, throttled, etc...).  
We do not have such code now but this is something we might want to implement (paired storage regions, HA endpoints in CosmosDB, etc...).

Also, for streaming requests, we could (theoretically) reuse the same struct at each request and just change the *next marker*. 

You proposed change to use `bytes` partially solves the issue since it's refcounted anyway.